src/main/java/com/conveyal/r5/profile/PerTargetPropagater.java

```diff
@@ -9,230 +12,204 @@
 import java.util.Arrays;
 
 /**
- * This class propagates from times at transit stops to times at destinations (targets). It is called with a function
- * that will be called with the target index (which is the row-major 1D index of the destination that is being
- * propagated to) and the array of whether the target was reached within the travel time cutoff in each Monte Carlo
- * draw. This is used in GridComputer to perform bootstrapping of accessibility given median travel time. This function
- * is only called for targets that were ever reached. It may seem needlessly generic to use a lambda function, but it
- * allows us to confine the bootstrapping code to GridComputer.
+ * Given minimum travel times from a single origin point to all transit stops, this class finds minimum travel times to
+ * a grid of destinations ("targets") by walking or biking or driving from the transit stops to the targets.
+ * We make one propagator instance per origin point.
  *
- * We propagate to a single target (grid cell) at a time because we only intend to store a few percentiles of travel time
- * at each target. Propagating the other direction, from stops out to targets, requires storing every travel time for
- * every MC draw/departure time, at every target. This would be impractically huge. To handle one target at a time we
- * need to invert the table of distances from stops to nearby targets: we instead use a table of distances from targets
- * to nearby stops.
+ * We find travel times to a single target (grid cell) at a time because we only intend to retain a few percentiles of
+ * travel time at each target, rather than retaining the travel time for every Monte Carlo schedule at every departure
+ * minute. Propagating the other direction (from stops out to targets) requires retaining every travel time for every
+ * Monte Carlo draw/departure time combination at every target. The resulting data structure would be impractically
+ * huge. To handle one target at a time rather than one stop at a time, we need to invert the table of distances from
+ * stops to their nearby targets: we instead use a table of distances to targets from their nearby stops.
+ *
+ * Note: this class is not threadsafe. It must process one target at a time sequentially in a single thread.
  */
 public class PerTargetPropagater {
 
     private static final Logger LOG = LoggerFactory.getLogger(PerTargetPropagater.class);
 
-    /** The travel time cutoff in this regional analysis */
+    // FIXME why are these fields public?
+
+    /** The travel time cutoff in this regional analysis FIXME why would this have a default value? */
     public int cutoffSeconds = 120 * 60;
```

Comment:
Is this the cutoff specified in the regional request (i.e. `task.maxTripDurationMinutes`), or the maximum possible cutoff used as an optimization here?

Reply:
Thanks for pointing that out. I always call the threshold travel time in the indicator travel time the "cutoff". Another place where an official glossary would be helpful. I have changed the comment.